include/rmm/mr/device/cuda_async_memory_resource.hpp

```diff
@@ -83,10 +83,9 @@ class cuda_async_memory_resource final : public device_memory_resource {
 
     // Allocate and immediately deallocate the initial_pool_size to prime the pool with the
     // specified size
-    if (initial_pool_size) {
-      auto p = do_allocate(*initial_pool_size, cuda_stream_default);
-      do_deallocate(p, *initial_pool_size, cuda_stream_default);
-    }
+    auto const pool_size = initial_pool_size.value_or(free * 0.5);
```

Comment:
Won't this assign a `double` (`free * 0.5`) to a `size_t`? Is there no compiler warning for that?

Reply:
Not by default. If it were in the context of braces (`{}`) constructing an object, then it would cause a warning.